libirods_rule_engine_plugin-elasticsearch.cpp

```diff
@@ -369,7 +421,13 @@ namespace
 				// clang-format off
 				ss << json{
 					{"absolutePath", _object_path},
+#ifdef IRODS_HAS_FEATURE_ADMIN_MODE_FOR_DSTREAM_LIBRARIES
 					{"data", std::string_view(buffer.data(), in.gcount())}
+#else
+					// Take the max to avoid passing an integer that's less than zero to the
+					// the string_view constructor.
+					{"data", std::string_view(buffer.data(), std::max(0, bytes_read))}
```

Comment:
Add a statement explaining how `bytes_read` can result in an integer less than zero.